action/protocol/rewarding/protocol.go

```diff
@@ -373,13 +375,20 @@ func (p *Protocol) createReceipt(
 	gasConsumed uint64,
 	logs ...*action.Log,
 ) *action.Receipt {
+	// remove possible nil log
+	nlogs := make([]*action.Log, 0, len(logs))
+	for _, l := range logs {
+		if l != nil {
+			nlogs = append(nlogs, l)
+		}
+	}
```

Comment:
better off to use []*action.Log as input, to avoid this hassle

like settleAction() in staking protocol

Reply:
*action.Log in []*action.Log slice can be still nil